src/tests/test_process.py

```diff
@@ -94,8 +144,42 @@ def _test_threads(self):
         #   - InvBroadcaster
         #   - AddrBroadcaster
         #   - Downloader
-        self.assertEqual(
-            len(self.process.threads()), self._threads_count)
+        #   - Uploader
+
+        self.longMessage = True
+
+        try:
+            thread_names = \
+                subprocess.check_output(["ps", "-L", "-o", "comm=",
+                                         "--pid",
+                                         str(self.process.pid)]).split()
+        # pylint: disable=broad-except
+        except BaseException:
+            thread_names = []
+
+        extra_threads = []
+        missing_threads = []
+        for thread_name in thread_names:
+            if thread_name not in self._threads_names:
+                extra_threads.append(thread_name)
+        for thread_name in self._threads_names:
+            if thread_name not in thread_names:
+                missing_threads.append(thread_name)
+
+        msg = "Missing threads: {}, Extra threads: {}".format(
+            ",".join(missing_threads), ",".join(extra_threads))
+
+        running_threads = len(self.process.threads())
+
+        self.assertGreaterEqual(
+            running_threads,
+            self._threads_count_min,
+            msg)
+
+        self.assertLessEqual(
+            running_threads,
+            self._threads_count_max,
+            msg)
```

Comment:
```patch
commit 77680a2ef4a9dd16736960e457529a66d7a6856d
Author: Dmitri Bogomolov <4glitch@gmail.com>
Date:   Mon Oct 26 07:57:52 2020 +0200

    Fix test_threads for non-linux environment:
    
     - check found thread_names adequacy, use num_threads()
     - take into account one extra thread on windows

diff --git a/src/tests/test_process.py b/src/tests/test_process.py
index cd114323..63aa2e9c 100644
--- a/src/tests/test_process.py
+++ b/src/tests/test_process.py
@@ -12,8 +12,6 @@ import unittest
 
 import psutil
 
-from sys import platform
-
 from pybitmessage.paths import set_appdata_folder
 
 
@@ -126,49 +124,36 @@ class TestProcessProto(unittest.TestCase):
             cls._cleanup_files()
 
     def _test_threads(self):
-        # only count for now
-        # because of https://github.com/giampaolo/psutil/issues/613
-        # PyBitmessage
-        #   - addressGenerator
-        #   - singleWorker
-        #   - SQL
-        #   - objectProcessor
-        #   - singleCleaner
-        #   - singleAPI
-        #   - Asyncore
-        #   - ReceiveQueue_0
-        #   - ReceiveQueue_1
-        #   - ReceiveQueue_2
-        #   - Announcer
-        #   - InvBroadcaster
-        #   - AddrBroadcaster
-        #   - Downloader
-        #   - Uploader
-
         self.longMessage = True
 
         try:
-            thread_names = \
-                subprocess.check_output(["ps", "-L", "-o", "comm=",
-                                         "--pid",
-                                         str(self.process.pid)]).split()
-        # pylint: disable=broad-except
-        except BaseException:
+            # using ps for posix platforms
+            # because of https://github.com/giampaolo/psutil/issues/613
+            thread_names = subprocess.check_output([
+                "ps", "-L", "-o", "comm=", "--pid",
+                str(self.process.pid)
+            ]).split()
+        except:  # pylint: disable=broad-except
             thread_names = []
 
-        extra_threads = []
-        missing_threads = []
-        for thread_name in thread_names:
-            if thread_name not in self._threads_names:
-                extra_threads.append(thread_name)
-        for thread_name in self._threads_names:
-            if thread_name not in thread_names:
-                missing_threads.append(thread_name)
-
-        msg = "Missing threads: {}, Extra threads: {}".format(
-            ",".join(missing_threads), ",".join(extra_threads))
-
-        running_threads = len(self.process.threads())
+        running_threads = len(thread_names)
+        if 0 < running_threads < 30:  # adequacy check
+            extra_threads = []
+            missing_threads = []
+            for thread_name in thread_names:
+                if thread_name not in self._threads_names:
+                    extra_threads.append(thread_name)
+            for thread_name in self._threads_names:
+                if thread_name not in thread_names:
+                    missing_threads.append(thread_name)
+
+            msg = "Missing threads: {}, Extra threads: {}".format(
+                ",".join(missing_threads), ",".join(extra_threads))
+        else:
+            running_threads = self.process.num_threads()
+            if sys.platform.startswith('win'):
+                running_threads -= 1  # one extra thread on Windows!
+            msg = "Unexpected running thread count"
 
         self.assertGreaterEqual(
             running_threads,
@@ -220,7 +205,6 @@ class TestProcess(TestProcessProto):
                 'Failed to read file %s' % pfile
             )
 
-    @unittest.skipIf(platform.startswith('darwin'), "Doesn't work on OSX")
     def test_threads(self):
         """Testing PyBitmessage threads"""
         self._test_threads()

```

Reply:
Almost clear: https://travis-ci.org/github/g1itch/PyBitmessage/builds/738955127

Reply:
I added this now.